SignalServiceKit/src/Messages/OWSIdentityManager.m

```diff
@@ -745,6 +747,58 @@ - (void)saveChangeMessagesForRecipientId:(NSString *)recipientId
     }];
 }
 
+#pragma mark - Debug
+
+#if DEBUG
+- (void)clearIdentityState
+{
+    [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+        NSMutableArray<NSString *> *identityKeysToRemove = [NSMutableArray new];
+        [transaction enumerateKeysInCollection:TSStorageManagerIdentityKeyStoreCollection
+                                    usingBlock:^(NSString *_Nonnull key, BOOL *_Nonnull stop) {
+                                        if ([key isEqualToString:TSStorageManagerIdentityKeyStoreIdentityKey]) {
+                                            // Don't delete our own key.
+                                            return;
+                                        }
+                                        [identityKeysToRemove addObject:key];
+                                    }];
+        for (NSString *key in identityKeysToRemove) {
+            [transaction removeObjectForKey:key inCollection:TSStorageManagerIdentityKeyStoreCollection];
+        }
+        [transaction removeAllObjectsInCollection:TSStorageManagerTrustedKeysCollection];
+    }];
+}
+
+- (NSString *)identityKeySnapshotFilePath
+{
+    NSString *dirPath = [OWSFileSystem appDocumentDirectoryPath];
+    return [dirPath stringByAppendingPathComponent:@".identity-key-snapshot"];
```

Comment:
Do we need to ensure these aren't backed up?

Reply:
OWSBackup's "full backups" backs up files and directories in the "app documents" and "shared data container", but ignores any top-level files or directories in those locations whose names start with a period ".".  

I've added comments to that effect everywhere I'm taking advantage of that.

Note that iOS itself also creates at least one file/directory in these containers which:

* can't/shouldn't be backed up (or otherwise accessed by apps).
* have a period "." prefix to their name.

So I've just extended that convention.

Reply:
To be clear, I'm wondering if we need to do something like `protectFileOrFolderAtPath` on these files